src/payload.rs

```diff
@@ -7,9 +8,140 @@ use core::ops::{Deref, DerefMut};
 // | SW USE                        |               ACTUAL DMA PART                                    |
 // | rssi - 1 byte | pipe - 1 byte | length - 1 byte | pid_no_ack - 1 byte | payload - 1 to 252 bytes |
 
+/// A builder for an `EsbHeader` structure
+///
+/// The builder is converted into an `EsbHeader` by calling the
+/// `check()` method.
+///
+/// ## Example
+///
+/// ```rust
+/// use esb::EsbHeaderBuilder;
+///
+/// let header_result = EsbHeaderBuilder::default()
+///     .max_payload(252)
+///     .pid(0)
+///     .pipe(0)
+///     .no_ack(true)
+///     .check();
+///
+/// assert!(header_result.is_ok());
+/// ```
+///
+/// ## Default Header Contents
+///
+/// By default, the following settings will be used:
+///
+/// | Field     | Default Value |
+/// | :---      | :---          |
+/// | pid       | 0             |
+/// | no_ack    | false         |
+/// | length    | 0             |
+/// | pipe      | 0             |
+///
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub struct EsbHeaderBuilder(EsbHeader);
+
+impl Default for EsbHeaderBuilder {
+    fn default() -> Self {
+        EsbHeaderBuilder(EsbHeader {
+            rssi: 0,
+            pid_no_ack: 0,
+            length: 0,
+            pipe: 0,
+        })
+    }
+}
```

Comment:
If the default value for `no_ack` is `false` then the default value for `pid_no_ack` must be `1` (assuming `pid == 0`).
Why Nordic uses `no-ack` as `active-low` ? No idea.

Reply:
Oof. Silly double negatives. Fixed.